crates/ruff_workspace/src/options.rs

```diff
@@ -2381,11 +2390,141 @@ impl PyUpgradeOptions {
+    ///   (e.g., the directory containing your `pyproject.toml`).
+    ///
+    /// For more information on the glob syntax, refer to the [`globset` documentation](https://docs.rs/globset/latest/globset/#syntax).
+    ///
+    /// Note that you don't need to list files that are excluded by
+    /// [`exclude`](#exclude).
+    // TODO(micha): Document once the CLI supports the exclude option
+    // #[option(
+    //     default = r#"[]"#,
+    //     value_type = "list[str]",
+    //     example = r#"
+    //         [format]
+    //         exclude = ["generated"]
+    //     "#
+    // )]
+    pub exclude: Option<Vec<String>>,
+
+    /// Whether to enable the unstable preview style formatting.
+    #[option(
+        default = "false",
+        value_type = "bool",
+        example = r#"
+            [format]
+            # Enable preview style formatting
+            preview = true
+        "#
+    )]
+    pub preview: Option<bool>,
+
+    /// Whether to use 4 spaces or hard tabs for indenting code.
+    ///
+    /// Defaults to 4 spaces. We only recommend changing this option, if you need to for accessibility reasons.
+    #[option(
+        default = "space",
+        value_type = r#""space" | "tab""#,
+        example = r#"
+            [format]
+            # Use tabs instead of 4 space indentation
+            indent-style = "tab"
+        "#
+    )]
+    pub indent_style: Option<IndentStyle>,
+
+    /// Whether to prefer single `'` or double `"` quotes for strings and docstrings.
+    ///
+    /// Ruff may deviate from this option if using the configured quotes would require more escaped quotes:
+    ///
+    /// ```python
+    /// a = "It's monday morning"
+    /// b = "a string without any quotes"
+    /// ```
+    ///
+    /// Ruff leaves `a` unchanged when using `quote-style = "single"` because it is otherwise
+    /// necessary to escape the `'` which leads to less readable code: `'It\'s monday morning'`.
+    /// Ruff changes the quotes of `b` to use single quotes.
+    #[option(
+        default = r#"double"#,
+        value_type = r#""double" | "single""#,
+        example = r#"
+            [format]
+            # Prefer single quotes over double quotes
+            quote-style = "single"
+        "#
+    )]
+    pub quote_style: Option<QuoteStyle>,
+
+    /// Ruff uses existing trailing commas as an indication that short lines should be left separate.
+    /// If this option is set to `true`, the magic trailing comma is ignored.
+    #[option(
+        default = r#"false"#,
+        value_type = r#"bool"#,
+        example = r#"
+            [format]
+            # Ignore magic trailing commas
+            skip-magic-trailing-comma = true
+        "#
+    )]
+    pub skip_magic_trailing_comma: Option<bool>,
+
+    /// The character Ruff uses at the end of a line.
+    #[option(
+        default = r#"lf"#,
+        value_type = r#""lf" | "crlf" | "auto" | "native""#,
```

Comment:
Can we mention what "auto" and "native" mean in the documentation? The "auto" is explained in the example so we could skip that but I would add it for consistency.

Reply:
Makes sense. It's a bit unfortunate because we already have the documentation on `LineEnding`.